webdriver/tests/support/fixtures_bidi.py

```diff
@@ -65,19 +65,17 @@ async def execute_as_async(sync_func, **kwargs):
 async def subscribe_events(bidi_session):
     subscriptions = []
 
-    async def subscribe_events(events, contexts=None):
-        result = await bidi_session.session.subscribe(events=events, contexts=contexts)
-        subscriptions.append((events, contexts))
+    async def subscribe_events(events, contexts=None, user_contexts=None):
+        result = await bidi_session.session.subscribe(events=events, contexts=contexts, user_contexts=user_contexts)
+        subscriptions.append(result["subscription"])
         return result
 
     yield subscribe_events
 
-    for events, contexts in reversed(subscriptions):
-        try:
-            await bidi_session.session.unsubscribe(events=events,
-                                                   contexts=contexts)
-        except (InvalidArgumentException, NoSuchFrameException):
-            pass
+    try:
+        await bidi_session.session.unsubscribe(subscriptions=subscriptions)
+    except (InvalidArgumentException, NoSuchFrameException):
```

Comment:
With this approach, if any subscription was removed by the test, then we will fail to remove the other ones because of the invalid argument error right?

Means there are more chances to leave subscriptions at the end of the test than the previous approach. Should we keep the loop + try / catch and unsubscribe each subscription independently?

Reply:
> With this approach, if any subscription was removed by the test, then we will fail to remove the other ones because of the invalid argument error right?

The error should happen only if the test unsubscribed by ID.

Reply:
Right but that's still possible. The fixture should still be able to unsubscribe from as many subscriptions as possible?